server/src/Server.cpp

```diff
@@ -281,6 +282,17 @@ Status Server::TestsGenServiceImpl::ProcessBaseTestRequest(BaseTestGen &testGen,
                            lineInfo, testsWriter, testGen.isBatched(), interactiveMode);
         auto finish_time = std::chrono::steady_clock::now();
         LOG_S(INFO) << "KLEE time: " << std::chrono::duration_cast<std::chrono::milliseconds>(finish_time - start_time).count() << " ms\n";
+
+        if (tryCompile) {
+            bool flag = true;
+            for (const auto &test: testGen.tests) {
+                  flag = flag && TestRunner::buildTest(testGen.projectContext, test.first);
+            }
+            if (!flag) {
+                return Status(StatusCode::FAILED_PRECONDITION,
+                              "Test build failed");
+            }
+        }
```

Comment:
You might want to move it to the test framework from the server. For example, it can be done by creating a parent of `TestsWriter` which verifies the content of written files

Reply:
Ok, i will do it.

Reply:
Move to build to TestRunner, because this check depends TestWriter and makefile from Linker

Reply:
Okay, that's much better